### DISCUSSION TITLE
Constant.pm module error following Perl 5.38 upgrade

### DISCUSSION THREAD

**Discussion by asser** [theory/sqitch#800] [Q&A]
Hi all,

Trying to run Sqitch after my local Perl installation got upgraded to 5.38 (running Arch Linux), I'm getting the following cryptic error:

```
Constants from lexical variables potentially modified elsewhere are no longer permitted at /usr/share/perl5/core_perl/constant.pm line 41.
BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/constant.pm line 45.
Compilation failed in require at /usr/lib/perl5/5.38/core_perl/File/Spec/Unix.pm line 115.
BEGIN failed--compilation aborted at /usr/lib/perl5/5.38/core_perl/File/Spec/Unix.pm line 115.
Compilation failed in require at /usr/lib/perl5/5.38/core_perl/File/Spec.pm line 21.
Compilation failed in require at /usr/share/perl5/core_perl/FindBin.pm line 86.
BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/FindBin.pm line 86.
Compilation failed in require at t/sqitch line 12.
BEGIN failed--compilation aborted at t/sqitch line 12.
```

I've tried reinstalling from the Arch package, with cpanminus, and now running directly from the latest git clone. Different versions (started with 1.1.0, then 1.2.0, 1.4.0 and finally git-latest).

Also tried to figure out by searching what would cause this error to get raised, and looked through the Github discussions and issues. No luck so far. It seems to be raised on import of the Perl constant module, but I can't see from the trace what exactly causes it.

Perldoc says this about the deprecation: https://perldoc.perl.org/perldeprecation#Constants-from-lexical-variables-potentially-modified-elsewhere

Is this an issue with Sqitch and Perl 5.38? Or am I missing something else? Where do I go from here to debug it? :)

**Comment by theory** [comment]
It looks like it's complaining about a [change from 5.32](https://perldoc.perl.org/perl5320delta#Modifiable-variables-are-no-longer-permitted-in-constants) on [line 115 of File::Spec::Unix](https://metacpan.org/dist/PathTools/source/lib/File/Spec/Unix.pm#L115), which looks innocuous to me. What version of File::Spec do you have?

**Comment by theory** [comment]
Oh, that appears to be in the Perl core modules. Very odd. I don't believe it's an issue in Sqithc itself, but maybe Arch linux has a bogus patch on the File::Spec::Unix module?

**Comment by asser** [comment]
Thanks for looking into it @theory ! Yea to me it also looks like some strange Perl core problem. With Sqitch 1.1.0 I get something different but similar in Getopt::Long (already in the first `use` line):

```
Constants from lexical variables potentially modified elsewhere are no longer permitted at /usr/share/perl5/core_perl/constant.pm line 41.
BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/constant.pm line 45.
Compilation failed in require at /usr/share/perl5/core_perl/Getopt/Long.pm line 222.
BEGIN failed--compilation aborted at /usr/share/perl5/core_perl/Getopt/Long.pm line 222.
Compilation failed in require at /usr/share/perl5/vendor_perl/App/Sqitch.pm line 9.
BEGIN failed--compilation aborted at /usr/share/perl5/vendor_perl/App/Sqitch.pm line 9.
Compilation failed in require at /usr/bin/vendor_perl/sqitch line 13.
BEGIN failed--compilation aborted at /usr/bin/vendor_perl/sqitch line 13.
```

So yea, it was mostly to see if someone in here had experienced something similar. I agree it looks more like a Perl installation problem with Arch. Will try to investigate more if I can.
